packages/arb-upgrades/index.ts

```diff
@@ -0,0 +1,361 @@
+  }> => {
+    const network = await hre.ethers.provider.getNetwork()
+    const path = `${rootDir}/deployments/${network.chainId}_queued-updates.json`
+    try {
+      const jsonBuff = readFileSync(path)
+      return { path, data: JSON.parse(jsonBuff.toString()) as QueuedUpdates }
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        console.log('New network; creating queued updates file')
+        writeFileSync(path, JSON.stringify({}))
+        return { path, data: {} }
+      } else {
+        throw err
+      }
+    }
+  }
+
+  const getDeployments = async (): Promise<{
+    path: string
+    data: CurrentDeployments
+  }> => {
+    const network = await hre.ethers.provider.getNetwork()
+    const path = `${rootDir}/deployments/${network.chainId}_current_deployment.json`
+    try {
+      const jsonBuff = readFileSync(path)
+      return {
+        path,
+        data: JSON.parse(jsonBuff.toString()) as CurrentDeployments,
+      }
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        console.log(
+          'New network; need to set up _current_deployments.json file'
+        )
+        throw err
+      } else {
+        throw err
+      }
+    }
+  }
+
+  const getBuildInfoString = async (contractName: string) => {
+    const contracts = (await hre.artifacts.getAllFullyQualifiedNames()).filter(
+      curr => curr.includes(contractName)
+    )
+    if (contracts.length !== 1) throw new Error('Contract not found')
+    const info = await hre.artifacts.getBuildInfo(contracts[0])
+    return JSON.stringify(info)
+  }
+
+  const deployLogic = async (
+    contractNames: ContractNames[] | ContractNames
+  ) => {
+    ensureCleanGitTree()
+
+    if (!Array.isArray(contractNames)) {
+      contractNames = [contractNames]
+    }
+    console.log('Deploying logic contracts for ', contractNames.join(','))
+    const signers = await hre.ethers.getSigners()
+    if (!signers.length) {
+      throw new Error(
+        'No signer - make sure a key is properly set (check hardhat config)'
+      )
+    }
+
+    const signer = signers[0]
+    console.log('Using signer', signer.address)
+
+    const { path, data: queuedUpdatesData } = await getQueuedUpdates()
+
+    for (const contractName of contractNames) {
+      console.log('Deploying new logic for ', contractName)
+
+      const contractFactory = (
+        await hre.ethers.getContractFactory(contractName)
+      ).connect(signer)
+      // TODO: ensure we're on the right layer?
+      const newLogic = await contractFactory.deploy()
+      const deployedContract = await newLogic.deployed()
+      const receipt = await deployedContract.deployTransaction.wait()
+
+      const newLogicData: QueuedUpdate = {
+        address: receipt.contractAddress,
+        deployTxn: receipt.transactionHash,
+        arbitrumCommitHash: currentCommit,
+        buildInfo: await getBuildInfoString(contractName),
+      }
+      queuedUpdatesData[contractName] = newLogicData
```

Comment:
before making the deployment, should we check if there is an update already queued for the current contract?
if so, im not sure we want to override the update (if so, we should surface some warning about it first)